lib/runner/cli/clirunner.js

```diff
@@ -348,13 +348,23 @@ CliRunner.prototype = {
       }
     }
 
-    if (envs.length > 1) {
+    if (envs.length > 1 && (this.settings.testWorkers == null || !this.settings.testWorkers.enabled)) {
       this.setupParallelMode(envs, done);
       return this;
     }
 
     this.setTestSettings(this.argv.env);
 
+    if (!this.isParallelMode() && envs.length < 2 && 
```

Comment:
this is very hard to follow and I would suggest creating a separate method instead.